test/services/AddressLookupServiceSpec.scala

```diff
@@ -0,0 +1,158 @@
+                                List(addressLine1, addressLine2, addressLine3, addressLine4),
+                                Some(postcode),
+                                Some(countryCode)
+                              ),
+        userAnswers = emptyUserAnswers.copy(warehouseList = warehouseMap))
+
+      res.warehouseList mustBe Map("1" -> Warehouse(Some("super cola"),
+                                          UkAddress(List("33 Rhes Priordy", "East London"), "E73 2RP")),
+                                   "12" -> Warehouse(Some(organisation),
+                                          UkAddress(List(addressLine1, addressLine2, addressLine3, addressLine4), postcode)))
+    }
+
+    "add to the cache the address of a packing site when a user returns from address lookup frontend with missing address lines" in {
+      val addressLookupState = Packingdetails
+      val packingMap = Map("1" -> Site(UkAddress(List("33 Rhes Priordy", "East London"), "E73 2RP"),Some("ref1"), Some("super cola"),None))
+      val AddedPackingSiteMissingLines = Map("1" -> Site(UkAddress(List("33 Rhes Priordy", "East London"), "E73 2RP"),Some("ref1"), Some("super cola"),None),"12" -> Site(UkAddress(List(addressLine1, addressLine2), postcode),None, Some(organisation),None))
+      val customerAddressMissingLines: AlfResponse = AlfResponse(
+        Some(organisation),
+        List(addressLine1, addressLine2),
+        Some(postcode),
+        Some(countryCode)
+      )
+
+      val res = service.addAddressUserAnswers(addressLookupState = addressLookupState,
+        address = customerAddressMissingLines,
+        userAnswers = emptyUserAnswers.copy(packagingSiteList = packingMap))
+
+      res.packagingSiteList mustBe AddedPackingSiteMissingLines
+    }
+
+    "add to the cache the address of a packing site when a user returns from address lookup frontend with full address lines" in {
+      val addressLookupState = Packingdetails
+      val packingMap = Map("1" -> Site(UkAddress(List("33 Rhes Priordy", "East London"), "E73 2RP"),Some("ref1"), Some("super cola"),None))
+      val AddedPackingSite = Map("1" -> Site(UkAddress(List("33 Rhes Priordy", "East London"), "E73 2RP"),Some("ref1"), Some("super cola"),None),"12" -> Site(UkAddress(List(addressLine1, addressLine2, addressLine3, addressLine4), postcode),None, Some(organisation),None))
+      val customerAddressMissingLines: AlfResponse = AlfResponse(
+        Some(organisation),
+        List(addressLine1, addressLine2, addressLine3, addressLine4),
+        Some(postcode),
+        Some(countryCode)
+      )
+
+      val res = service.addAddressUserAnswers(addressLookupState = addressLookupState,
+        address = customerAddressMissingLines,
+        userAnswers = emptyUserAnswers.copy(packagingSiteList = packingMap))
+
+      res.packagingSiteList mustBe AddedPackingSite
+    }
+
+    "don't add to userAnswers when no details are added in alf and throw exception" in {
```

Comment:
can we have test for packing details aswell

Reply:
This is testing packing details -> line 119

Reply:
Packing details has been done on lines 118 and 100